rfcs/docker_github_registry.md

```diff
@@ -0,0 +1,41 @@
+# RFC 198: Move the container image from Docker Hub to GitHub Registry
+
+## Summary
+
+This RFC proposes moving the container image from Docker Hub to GitHub Registry.
+
+## Details
+
+This RFC addresses the issue
+[#28903](https://github.com/web-platform-tests/wpt/issues/28903). Docker Hub
+organization allows for a limited number of members and only a few members can
+publish a new Docker image that sometimes results in the Docker image that is
+used in taskcluster being outdated as documented in [this PR](https://github.com/web-platform-tests/wpt/pull/44576#issue-2133724788).
+
+The following are the benefits of using the GitHub registry for hosting:
+
+- It is [hosted on GitHub](https://github.com/web-platform-tests/wpt/packages) together with the wpt repository.
+- Publishing is possible by any project member who has the write permissions to the
+  repository.
```

Comment:
I don't know what this image is used for, but that's a lot of people who we haven't (and probably don't want to) vet for more than "is willing to review tests"

Reply:
<img width="1113" alt="Screenshot 2024-05-29 at 10 19 34" src="https://github.com/web-platform-tests/rfcs/assets/399150/c3744c06-2f0a-49f5-a77a-0cdb76b5e7d8">

Members with the Write access can make changes to the Dockerfile so it seems appropriate to also allow publishing the changes. Alternatively, on the package settings page there is an option to not inherit the permissions or disallow access from the workflows so the admin can change that and only allow selected members if that is desired.

Reply:
We could use a different repo for the images if we wanted different permissions. 

We can also use `CODEOWNERS` to ensure that any changes to the docker image get admin review before being merged which would prevent updating the CI to use a new version of the image, but wouldn't prevent  publishing images in the first place.

We can also change the permissions on specific packages (notably the `wpt` package) to not inherit from the wpt base permissions. I think it would still (and maybe always was?) possible for people to publish new packages which would by default inherit permissions from the repo.